components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -139,8 +142,12 @@ public Response get(@PathParam("id") String id) {
     @Path("/")
     public Response create(DashboardMetadata dashboardMetadata) {
         try {
-            dashboardDataProvider.add(dashboardMetadata);
-            return Response.status(CREATED).build();
+            if (dashboardDataProvider.get(dashboardMetadata.getUrl()).equals(Optional.empty())) {
```

Comment:
Improper way to consume an Optional. Use isPresent() method instead.

Reply:
BTW, shall we add a new method `exists(String dashboardUrl)` to `DashboardMetadataProvider` interface? Since we only need to check the existence of a dashboard for a given URL here, retrieving the dashboard's data is unnecessary &  it hurts performance.